File: fsharp/scrabble-score/ScrabbleScore.fs
module ScrabbleScore
open System

let scoreMapText = 
    @"A, E, I, O, U, L, N, R, S, T     1
    D, G                               2
    B, C, M, P                         3
    F, H, V, W, Y                      4
    K                                  5
    J, X                               8
    Q, Z                               10" 
let scoreMap = 
    let processRow row =
        let value = Array.head row |> int
        Array.tail row |> Array.map (fun c -> (c, value))
    let cells = 
        scoreMapText.Split('\n', StringSplitOptions.RemoveEmptyEntries) 
        |> Array.map (fun line -> line.Replace(",", "").Split(' ', StringSplitOptions.RemoveEmptyEntries) |> Array.rev |> processRow)
    cells |> Array.collect id |> Map.ofArray

let score (word:string) = Seq.sumBy (fun x -> scoreMap.[x.ToString().ToUpper()]) word

File: python/scrabble-score/scrabble_score.py

rules = """
    A, E, I, O, U, L, N, R, S, T       1
    D, G                               2
    B, C, M, P                         3
    F, H, V, W, Y                      4
    K                                  5
    J, X                               8
    Q, Z                               10"""

scores = dict()
for line in rules.split('\n'):
    letters = line.replace(',', '').split()
    for i in range(0, len(letters)):
        scores[letters[i]] = int(letters[-1])

def score(word):
    return sum(scores[c] for c in word.upper())
